docs/source/conf.py

```diff
@@ -85,11 +97,108 @@
 
 katex_prerender = True
 
+# General information about the project.
+project = "PyTorch"
+copyright = "PyTorch Contributors"
+author = "PyTorch Contributors"
+torch_version = str(torch.__version__)
+
+# The version info for the project you're documenting, acts as replacement for
+# |version| and |release|, also used in various other places throughout the
+# built documents.
+#
+# The short X.Y version.
+# TODO: change to [:2] at v1.0
+version = "main (" + torch_version + " )"
+# The full version, including alpha/beta/rc tags.
+# TODO: verify this works as expected
+release = "main"
+
+# Customized html_title here.
+# Default is " ".join(project, release, "documentation") if not set
+if RELEASE:
+    # Turn 1.11.0aHASH into 1.11
+    # Note: the release candidates should no longer have the aHASH suffix, but in any
+    # case we wish to leave only major.minor, even for rc builds.
+    version = ".".join(torch_version.split(".")[:2])
+    # html_title = " ".join((project, version, "documentation"))
+    release = version
+
+switcher_version = "main" if not RELEASE else version
+html_static_path = ["_static"]
+html_theme_options = {
+    "logo": {"text": "Home"},
+    "analytics_id": "GTM-T8XT4PS",
+    "canonical_url": "https://pytorch.org/docs/stable/",
+    "switcher": {
+        "json_url": "https://pytorch.org/docs/pytorch-versions.json",
+        "version_match": switcher_version,
+    },
+    "navigation_with_keys": False,
+    "external_links": [
+        {
+            "name": "Tutorials",
+            "url": "https://pytorch.org/tutorials/",
+        },
+    ],
+    "show_version_warning_banner": True,
```

Comment:
What is that?

Reply:
This banner appears if you are viewing not a "preferred" branch, that is in our case "stable". This is the banner:

![warning-banner-version](https://github.com/user-attachments/assets/9e230ca0-5d6a-4704-8823-802d640d73f1)